server/routerlicious/packages/services-core/src/checkpointService.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * A service to manage common checkpoint operations in scribe and deli
+ */
+
+import {
+	BaseTelemetryProperties,
+	getLumberBaseProperties,
+	LumberEventName,
+	Lumberjack,
+} from "@fluidframework/server-services-telemetry";
+import { ICheckpointRepository, IDocumentRepository } from "./database";
+import { IDeliState, IDocument, IScribe } from "./document";
+
+export class CheckpointService implements ICheckpointService {
+	constructor(
+		private readonly checkpointRepository: ICheckpointRepository,
+		private readonly documentRepository: IDocumentRepository,
+		private readonly isLocalCheckpointEnabled: boolean,
+	) {}
+	localCheckpointEnabled: boolean = this.isLocalCheckpointEnabled;
+
+	async writeCheckpoint(
+		documentId: string,
+		tenantId: string,
+		service: string,
```

Comment:
nit: Can we restrict this to "deli" | "scribe"..we could do that for the other methods in this file too

`
type DocumentLambdas = 'deli' | 'scribe';
`